test/integration/run_backup.py

```diff
@@ -168,3 +163,36 @@ def backup_and_restore(filler, preparer, checker, env={}, backup_opts=None,
 			raise
 	as_srv.reset_aerospike_servers()
 
+def run_backup_w_valgrind(filler, context={}, backup_options=None):
+	"""
+	Run asbackup command with given options using valgrind
+	"""
+	as_srv.start_aerospike_servers()
+	filler(context)
+	try:
+		with open("val_bc.log", "w") as pipe_stdout:
+			lib.run("asbackup", *backup_options, do_async=False,
+				pipe_stdout=pipe_stdout, env={}, USE_VALGRIND=True)
+		res = lib.parse_val_logs("val_bc.log")
+	except:
+		as_srv.reset_aerospike_servers()
+		raise
+	as_srv.reset_aerospike_servers()
+	return res
```

Comment:
Suggestion: you could use `try`, `except`, `finally `here, so that `as_srv.reset_aerospike_servers()` only appears in the finally block.

Reply:
I see. `raise` will terminate the flow of the program and I dont think it will hit `finally` in this case.